engine/src/main/java/org/terasology/engine/core/modes/loadProcesses/CreateWorldEntity.java

```diff
@@ -52,64 +55,101 @@ public String getMessage() {
 
     @Override
     public boolean step() {
+        if (worldEntityExists()) {
+            EntityRef worldEntity = getExistingWorldEntity();
+            configureChunkProvider(worldEntity);
+            useConfigurationOfCurrentWorld(worldEntity);
+        } else {
+            EntityRef worldEntity = createWorldPoolsAndEntity();
+            configureChunkProvider(worldEntity);
+            WorldConfigurator worldConfigurator = useConfigurationFromConfig();
+            configurateWorldEntity(worldConfigurator, worldEntity);
```

Comment:
I think I would opt to combine these two methods. useConfigurationFromConfig is just retrieving the config from context.

Reply:
You've discovered what I was aiming at during this refactoring!

Yes, my goal was to have a function that provides the appropriately configured WorldConfigurator, so I could figure out what would be different when changing where it gets that information as in #4531.